### ISSUE TITLE
`TIME_PARSE()` works for constants, fails for columns

### ISSUE THREAD

**Issue by paul-rogers** [apache/druid#11660]
### Affected Version

Version: 0.21.1

### Description

SQL Druid provides the [`TIME_PARSE()`](https://druid.apache.org/docs/latest/querying/sql.html#time-functions) function.

Let's try it against the `segments` table. First, verify the data, on a system with only the sample Wikipedia data:

```sql
SELECT
  "start"
FROM sys.segments
LIMIT 1
```

Result:
```text
2015-09-12T00:00:00.000Z
```

Now, use `TIME_PARSE()` with the returned value. (OK, the query is kind of lame, but bear with me):

```sql
SELECT
  TIME_PARSE('2015-09-12T00:00:00.000Z')
FROM sys.segments
LIMIT 1
```

Result:
```text
2015-09-12T00:00:00.000Z
```

Now, put the two together: we know the `start` column has a perfectly fine timestamp string. Try to parse it:

```sql
SELECT
  TIME_PARSE("start")
FROM sys.segments
LIMIT 1
```

This fails:

```text
Error: Unknown exception
cannot translate call TIME_PARSE($t2)
java.lang.RuntimeException
```

Expected that a SQL function would work just as well for columns as for constants.

Note: I was trying to use `TIME_PARSE()` to work around [another issue](https://github.com/apache/druid/issues/11659) that prevents using `MAX()` on a string.

**Comment by FrankChen021** [comment]
This might be a bug

**Comment by bananaaggle** [comment]
In document [sql](https://druid.apache.org/docs/latest/querying/sql.html), you can see this: 
`Note: "sys" tables do not currently support Druid-specific functions like TIME_PARSE and APPROX_QUANTILE_DS. Only standard SQL functions can be used.`

**Comment by FrankChen021** [comment]
@bananaaggle  From users' view, it's hard to understand why these function can't work on 'sys' tables. IMO, these implementation constaints should be removed in future.

**Comment by bananaaggle** [comment]
> @bananaaggle From users' view, it's hard to understand why these function can't work on 'sys' tables. IMO, these implementation constaints should be removed in future.

Yes. But I guess a query to data is different from query to metadata, they can not be adopted to each other easily.

**Comment by sirianni** [comment]
Is there a workaround for using any sort of date functions on the `sys` tables?

**Comment by itsautfullday** [comment]
> I just discovered the following syntax that works for converting varchar to timestamps for sys tables.
> 
> ```
> select cast(REPLACE(created_time,'T',' ') as TIMESTAMP) from sys.tasks limit 10
> ```

THE LEGEND! Was looking for this today in the morning and couldn't find anything and then in the afternoon found this answer. Totally worked! Thanks!

**Comment by JRobTS** [comment]
CAST no longer works as of Druid 29.0.1

Also not working: LEFT, RIGHT, SUBSTR, SUBSTRING
